### PR TITLE
Get the terminated pod by name

### LINKED ISSUE #13625: "kubectl get pods NAME" couldn't get the terminated pod
When the pod terminated, using "kubectl get pods NAME" couldn't get it.

### PR DESCRIPTION
**Pull request by feihujiang**
Fixes #13625

### CODE REVIEW

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -492,6 +492,10 @@ func translateTimestamp(timestamp util.Time) string {
 }
 
 func printPod(pod *api.Pod, w io.Writer, withNamespace bool, wide bool, showAll bool, columnLabels []string) error {
+	return printPodIsShowNecessarily(pod, w, withNamespace, wide, showAll, true, columnLabels)
+}
+
+func printPodIsShowNecessarily(pod *api.Pod, w io.Writer, withNamespace bool, wide bool, showAll bool, showNecessarily bool, columnLabels []string) error {
 	name := pod.Name
```
Comment (smarterclayton) [inline pkg/kubectl/resource_printer.go]: Would suggest instead of "ShowNecessarily" I would call this "printPodBase"

Comment (smarterclayton) [inline pkg/kubectl/resource_printer.go]: And make the variable "showIfTerminating"

**Comment by feihujiang** [comment]
@smarterclayton I updated this commit based on your feedback, please review again

**Comment by yujuhong** [comment]
Reassigned to @smarterclayton since he's already reviewed it.

File: pkg/kubectl/resource_printer.go [hunk 2]
```diff
@@ -504,6 +504,10 @@ func translateTimestamp(timestamp util.Time) string {
 }
 
 func printPod(pod *api.Pod, w io.Writer, withNamespace bool, wide bool, showAll bool, columnLabels []string) error {
+	return printPodBase(pod, w, withNamespace, wide, showAll, true, columnLabels)
+}
+
+func printPodBase(pod *api.Pod, w io.Writer, withNamespace bool, wide bool, showAll bool, showIfTerminating bool, columnLabels []string) error {
```
Comment (smarterclayton) [inline pkg/kubectl/resource_printer.go]: I didn't see showAll before - what does it do differently than showIfTerminating?

Comment (smarterclayton) [inline pkg/kubectl/resource_printer.go]: Is it end user indicated?

Comment (feihujiang) [inline pkg/kubectl/resource_printer.go]: Yes, end user indicate it, the command option is "-a, --show-all[=false]: When printing, show all resources (default hide terminated pods.)"  When a pod named test is terminated, "kubectl get pods test" doesn't show anything, "kubectl get pods test -a" could show it. But I don't think "kubectl get pods test -a" is easy to use. I think "kubectl get pods test" should show it.

**Comment by feihujiang** [comment]
When a pod named test is terminated, "kubectl get pods test" doesn't show anything, "kubectl get pods test -a" could show it. But I don't think "kubectl get pods test -a" is easy to use. I think "kubectl get pods test" should show it. So I new this issue to fix this.

File: pkg/kubectl/resource_printer.go (hunk 2 above)
Comment (yujuhong) [inline pkg/kubectl/resource_printer.go]: Agreed that `kubectl get pod <pod_name>` should retrieve the pod regardless of its state.
The `-a` was introduced relatively recently to mimic the behavior of `docker ps -a`. By default, `get pods` would hide all terminated pods.

**Comment by bgrant0607** [comment]
I agree that --show-all should not be necessary for explicitly named resources, specified either directly on the command line or via `-f`
cc @JanetKuo

**Comment by feihujiang** [comment]
@smarterclayton  Is there anything I should do about this commit?

**Comment by smarterclayton** [comment]
This is LGTM - we can come back to this w.r.t. the structure of hiding terminated pods.

**Pull request merged**